src/common/options/global.yaml.in

```diff
@@ -4000,15 +4000,39 @@ options:
   default: false
   with_legacy: true
 - name: bdev_enable_discard
+  desc: send discards to the block device
   type: bool
   level: advanced
   default: false
   with_legacy: true
+  flags:
+  - runtime
+  see_also:
+  - bdev_async_discard
+  - bdev_async_discard_threads
 - name: bdev_async_discard
-  type: bool
+  desc: send discards to the block device in one or more threads
+  type: uint
   level: advanced
   default: false
-  with_legacy: true
+  with_legacy: false
+  flags:
+  - runtime
+  see_also:
+  - bdev_enable_discard
+  - bdev_async_discard_threads
+- name: bdev_async_discard_threads
```

Comment:
We can permit 0 for this parameter and enforce sync discard mode this way. As a result we can get rid off bdev_async_discard parameter.
We need to fix try_discard and handle_conf_change then.
Not that I'm insisting on that but IMO the less parameters the better. What do you think?

Reply:
I agree, I'll take a quick pass at this.  It looks like I've already removed all use of `bdev_async_discard` but it's still defined anyway, so at a minimum that should be removed (and the comment in `::_queue_discard()` to be updated).

Reply:
This is complete with my last push, I think.  It's been a few days since I looked at this, so I may be foggy, but I think `::try_discard()` should be okay here.